shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/RouteDetailsStopList.kt

```diff
@@ -74,11 +74,16 @@ data class RouteDetailsStopList(val segments: List<Segment>) {
         suspend fun fromPieces(
             routeId: String,
             directionId: Int,
-            routeStops: RouteStopsResponse?,
+            routeStops: RouteStopsResult?,
             globalData: GlobalResponse,
         ): RouteDetailsStopList? =
             withContext(Dispatchers.Default) {
-                if (routeStops == null) return@withContext null
+                if (
+                    routeStops == null ||
+                        routeStops.routeId != routeId ||
+                        routeStops.directionId != directionId
```

Comment:
question(non-blocking): this seems reasonable to me, but I don't quite understand why an out of date result was being used (and am having trouble replicating myself). Is this a race condition when switching back and forth really quickly? If there is a mismatch, is there a possibility that this will indefinitely return null and the correct data will never be shown?

Reply:
> In theory the LaunchedEffect should be cancelled every time the state changes and it should always reload from the endpoint for the current state, regardless of the current state of the request.

Ah yep good call, that makes me feel better!